src/haxelib/Util.hx

```diff
@@ -28,33 +28,41 @@ class Util {
 
 	macro static public function getHaxelibVersionLong() {
 		var version:String = readVersionFromHaxelibJson();
+		// check if the .git folder exist
+		// prevent getting the git info of a parent directory
+		if (!sys.FileSystem.isDirectory(".git"))
+			return macro $v{version};
+
 		var p;
 		try {
-			//check if the .git folder exist
-			//prevent getting the git info of a parent directory
-			if (!sys.FileSystem.isDirectory(".git"))
-				throw "Not a git repo.";
-
 			//get commit sha
 			p = new sys.io.Process("git", ["rev-parse", "HEAD"]);
-			final sha = p.stdout.readAll().toString().trim();
+			var sha = p.stdout.readAll().toString().trim();
```

Comment:
Huh?

Reply:
That file got moved into the top level `haxelib` package, which causes it to get picked up when the server is compiling (which requires haxe 3).

I'm not sure why it wasn't failing before, but now it did:
https://github.com/HaxeFoundation/haxelib/runs/5971785204?check_suite_focus=true